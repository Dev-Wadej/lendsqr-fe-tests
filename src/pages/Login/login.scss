                @font-face {
                    font-family: "Avenir";
                    src: local("Avenir"), url("../../data/font/AvenirNextLTPro-Regular.otf") format("opentype");
                }
                
                @font-face {
                    font-family: "AvenirLight";
                    src: local("AvenirLight"), url("../../data/font/AvenirNextLTPro-It.otf") format("opentype");
                }
                
                @font-face {
                    font-family: "AvenirBold";
                    src: local("AvenirBold"), url("../../data/font/AvenirNextLTPro-Bold.otf") format("opentype");
                }
                
                .login-page {
                    & {
                        @media only screen and (min-width: 1800px) {
                            & {
                                display: grid;
                                place-content: center;
                                height: 100vh
                            }
                        }
                    }
                    font-family: 'Avenir';
                    width: 90vw;
                    margin-left: 5vw;
                    aside {
                        margin: 5rem 0 3rem 0rem;
                    }
                }
                
                .login-content {
                    margin-top: 6rem;
                    display: flex;
                    gap: 5rem;
                    @media only screen and (max-width: 1000px) {
                        & {
                            flex-direction: column;
                            justify-content: center;
                            align-items: center;
                            margin-top: 4rem;
                            gap: 3rem;
                        }
                    }
                    @media only screen and (max-width: 1000px) {
                        & {
                            margin-top: 2.5rem;
                            gap: 1.5rem;
                        }
                    }
                }
                
                .login-img {
                    width: 40rem;
                    @media only screen and (max-width: 1000px) {
                        & {
                            width: 30rem
                        }
                    }
                    @media only screen and (max-width: 1000px) {
                        & {
                            width: 20rem;
                        }
                    }
                    img {
                        width: 100%;
                        height: 100%;
                    }
                }
                
                .login-typography {
                    h1 {
                        font-family: 'AvenirBold';
                        font-weight: 700;
                        font-size: 1.7rem;
                        color: rgba(33, 63, 125, 1);
                        margin-bottom: .6rem;
                    }
                    h5 {
                        color: rgba(84, 95, 125, .9);
                        font-size: 1rem;
                        word-spacing: 110%;
                        font-weight: lighter;
                        margin-bottom: 3rem;
                    }
                }
                
                dl {
                    input {
                        display: block;
                        width: 23rem;
                        padding: .7rem;
                        margin-top: 1.5rem;
                        border: 1.5px solid rgba(84, 95, 125, 0.15);
                        border-radius: .3rem;
                    }
                    input::placeholder {
                        color: rgba(84, 95, 125, .6);
                        font-family: 'Avenir';
                    }
                }
                
                button {
                    border: none;
                    display: block;
                    cursor: pointer;
                }
                
                .forgot-password {
                    color: rgba(57, 205, 204, 1);
                    background-color: #fff;
                    word-spacing: .15rem;
                    margin: 1.2rem 0;
                    font-family: 'AvenirBold';
                    font-size: .62rem;
                }
                
                .login-btn {
                    color: #fff;
                    background-color: rgba(57, 205, 204, 1);
                    padding: .7rem 10.3rem;
                    border-radius: .3rem;
                    margin-top: 2rem;
                    font-family: 'AvenirBold';
                    font-size: .9rem;
                }
                
                input:focus,
                button:focus {
                    outline: none;
                }
                
                .show-password {
                    position: relative;
                    a {
                        position: absolute;
                        right: 1.4rem;
                        top: .9rem;
                        font-size: .63rem;
                        font-family: 'AvenirBold';
                        color: #39CDCC;
                    }
                }