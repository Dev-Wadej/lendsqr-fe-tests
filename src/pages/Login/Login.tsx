import { truncate } from 'fs/promises'
import React,{useState} from 'react'
import { useNavigate } from 'react-router-dom'

import loginPage from '../../data/images/loginPage.jpg'
import {ReactComponent as CompanyLogo} from '../../data/svgs/companyLogo.svg'
import './login.scss'

const Login = () => {

  const navigate= useNavigate()
  const [password,setPasword]=useState('')
  const [email,setEmail]= useState('')
  const [showPassword,setShowPassword]=useState(false)
  const [statusEmail,setStatusEmail]= useState(true)
  const [statusPassword,setStatusPassword]= useState(true)

  const handleShowPassWord=()=>{
    setShowPassword(prev=>!prev)
  }
  const handleLogin=()=>{
    if(!email){
      setStatusEmail(false)
      return;
    }
    if(!password){
      setStatusPassword(false)
      return;
    }
    if(email && password){
      navigate('./dashboard/user')
    }
  }
  const handleFormSubmit=(e:React.FormEvent)=>{
    e.preventDefault()
  }
  const handleEmailAuth=(e:React.ChangeEvent<HTMLInputElement>)=>{
    setEmail(e.target.value)
    setStatusEmail(true)
  }
    
  const hanldePasswordValue=(e: React.ChangeEvent<HTMLInputElement>)=>{
    setPasword(e.target.value)
    setStatusPassword(true)
  }

  return (
    <section className='login-page'>
      <aside>
        <CompanyLogo/>
      </aside>
      <div className='login-content'>
        <div  className='login-img'>
          <img src={loginPage} alt="Sign in Page Illustrator" />
        </div>
        <form className='login-typography' onSubmit={handleFormSubmit}>
          <h1>Welcome!</h1>
          <h5>Enter details to login</h5>
          <dl>
            <input type="email" placeholder='Email' name='email' onChange={handleEmailAuth} />
            {!statusEmail && <p className='email-validator'>Email is required</p>}

            <div className='show-password'>
            <input name='password' type={showPassword ? 'text' :'password'} placeholder='Password' onChange={hanldePasswordValue} value={password} />
            <a  href="#" onClick={handleShowPassWord}> {showPassword ? 'HIDE' : 'SHOW'} </a>
            </div>
            {!statusPassword && <p className='password-validator'>Enter an password</p>}
            
          </dl>
          <button className='forgot-password'>FORGOT PASSWORD?</button>
          <button type='submit' className='login-btn' onClick={handleLogin}>LOGIN</button>
        </form>
      </div>
      
    </section>
  )
}

export default Login
