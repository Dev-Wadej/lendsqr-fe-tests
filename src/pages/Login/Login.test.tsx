import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import userEvent from "@testing-library/user-event";
import Login from "./Login";
import renderer from 'react-test-renderer';



test('submitting the form',()=>{
    render(
        <MemoryRouter>
            <Login/>
        </MemoryRouter>
    )
    userEvent.type(screen.getByPlaceholderText(/email/i),'Joshua')
    userEvent.type(screen.getByPlaceholderText(/password/i),'i need none')
    userEvent.click(screen.getByRole('button',{name:/login/i}))
    screen.debug()

    const output = renderer.create(<MemoryRouter>
        <Login/>
    </MemoryRouter>  ).toJSON();
  expect(output).toMatchSnapshot();
})