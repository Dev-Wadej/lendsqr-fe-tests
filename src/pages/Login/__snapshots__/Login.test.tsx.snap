// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`submitting the form 1`] = `
<section
  className="login-page"
>
  <aside>
    <svg>
      companyLogo.svg
    </svg>
  </aside>
  <div
    className="login-content"
  >
    <div
      className="login-img"
    >
      <img
        alt="Sign in Page Illustrator"
        src="loginPage.jpg"
      />
    </div>
    <form
      className="login-typography"
      onSubmit={[Function]}
    >
      <h1>
        Welcome!
      </h1>
      <h5>
        Enter details to login
      </h5>
      <dl>
        <input
          name="email"
          placeholder="Email"
          type="text"
        />
        <div
          className="show-password"
        >
          <input
            name="password"
            onChange={[Function]}
            placeholder="Password"
            type="password"
            value=""
          />
          <a
            href="#"
            onClick={[Function]}
          >
             
            SHOW
             
          </a>
        </div>
      </dl>
      <button
        className="forgot-password"
      >
        FORGOT PASSWORD?
      </button>
      <button
        className="login-btn"
        onClick={[Function]}
        type="submit"
      >
        LOGIN
      </button>
    </form>
  </div>
</section>
`;
